ACE/ace/Functor.h

```diff
@@ -200,31 +200,27 @@ class ACE_Export ACE_Hash<unsigned long>
   unsigned long operator () (unsigned long t) const;
 };
 
-#if (ACE_SIZEOF_LONG < 8)
 /**
- * @brief Function object for hashing a signed 64-bit number
+ * @brief Function object for hashing a signed long long number
  */
 template<>
-class ACE_Export ACE_Hash<ACE_INT64>
+class ACE_Export ACE_Hash<long long>
```

Comment:
Why remove ACE_INT64? At the moment you are using int64_t the specialization is for int64_t in the original case, now you change it to long long

Reply:
Perhaps it is not the right thing to do, but you get the following error:
  
```In file included from /Users/wilsonj/OpenDDS/ACE_TAO/ACE/protocols/ace/RMCast/Acknowledge.cpp:7:
In file included from /Users/wilsonj/OpenDDS/ACE_TAO/ACE/protocols/ace/RMCast/Acknowledge.h:6:
In file included from /Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager.h:25:
In file included from /Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager_T.h:22:
In file included from /Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Functor_T.h:180:
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Functor_T.inl:28:11: error: member reference base type
      'const unsigned long long' is not a structure or union
  return t.hash ();
         ~^~~~~
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager_T.inl:78:10: note: in instantiation of member function
      'ACE_Hash<unsigned long long>::operator()' requested here
  return this->hash_key_ (ext_id);
         ^
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager_T.cpp:325:15: note: in instantiation of member function
      'ACE_Hash_Map_Manager_Ex<unsigned long long, ACE_RMCast::Acknowledge::Descr, ACE_Hash<unsigned long long>,
      ACE_Equal_To<ACE_UINT64>, ACE_Null_Mutex>::hash' requested here
  loc = this->hash (ext_id) % this->total_size_;
              ^
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager_T.cpp:234:13: note: in instantiation of member function
      'ACE_Hash_Map_Manager_Ex<unsigned long long, ACE_RMCast::Acknowledge::Descr, ACE_Hash<unsigned long long>,
      ACE_Equal_To<ACE_UINT64>, ACE_Null_Mutex>::shared_find' requested here
  if (this->shared_find (ext_id, entry, loc) == -1)
            ^
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager_T.inl:94:16: note: in instantiation of member function
      'ACE_Hash_Map_Manager_Ex<unsigned long long, ACE_RMCast::Acknowledge::Descr, ACE_Hash<unsigned long long>,
      ACE_Equal_To<ACE_UINT64>, ACE_Null_Mutex>::bind_i' requested here
  return this->bind_i (ext_id, int_id, temp);
               ^
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/ace/Hash_Map_Manager_T.inl:103:16: note: in instantiation of member function
      'ACE_Hash_Map_Manager_Ex<unsigned long long, ACE_RMCast::Acknowledge::Descr, ACE_Hash<unsigned long long>,
      ACE_Equal_To<ACE_UINT64>, ACE_Null_Mutex>::bind_i' requested here
  return this->bind_i (ext_id, int_id);
               ^
/Users/wilsonj/OpenDDS/ACE_TAO/ACE/protocols/ace/RMCast/Acknowledge.h:150:22: note: in instantiation of member
      function 'ACE_Hash_Map_Manager_Ex<unsigned long long, ACE_RMCast::Acknowledge::Descr, ACE_Hash<unsigned long
      long>, ACE_Equal_To<ACE_UINT64>, ACE_Null_Mutex>::bind' requested here
        int r (Base::bind (sn, d));
```
on ```clang --version
Apple clang version 11.0.0 (clang-1100.0.20.17)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin```
because sizeof(long)==8 which means that ACE_Hash is not defined for the 64-bit types which are long long and unsigned long long.